### ISSUE TITLE
Pan to Context does not work when default lat-long are set

### ISSUE THREAD

**Issue by worthingtonte** [ivosturm/GoogleMapsCustomMarker#2]
Hi,

I'm having an issue where if I have Pan to Context set as Yes, and a default lat-long value, then Pan to Context no longer works.

This is also an issue in the GoogleMaps widget, however, this can be worked around by setting default values of the attributes (to get the desired effect). Unfortunately, I cannot do this with this widget.

Thanks in advance,

Tom

**Comment by ivosturm** [comment]
Hi Tom,

Can you describe the scenario in which you are seeing this error? I can not reproduce this. So some info on the involved usecase would help. Is it a new object (not yet geocoded, not having lat long) for which you want to have a default lat long?

**Comment by worthingtonte** [comment]
Hi Ivo,

The scenario is that I require the map to have a default lat and lon, but also for Pan to context to work (when a new address is added).

In the standard Google Maps widget this works by setting the lat and lon default value on the attribute level (not by setting the widget default lat on lon, which would be ideal but this is an ok workaround). So im creating a new object with the lat and lon already set on the object.

Hope this helps,

Tom

**Comment by ivosturm** [comment]
Hi Tom,

It seems you want to use both features in the same map/screen? I am probably missing something since it does not sound logical to me. The 'Pan to context' feature you would need in a listen-to-grid dataview, listenening to a grid of already created locations. This can also be done via the XPath **[id='[%CurrentObject%]']** in stead of using 'Pan to context'.

The default lat lon, I am assuming is only needed in a new screen, so when creating a new location, not being in an overview with a listen to grid dataview.

Did you try the  **[id='[%CurrentObject%]']** XPath and not using the 'Pan to context' setting? If so and not helping you, then it is probably best to email me (ivo.sturm@firstconsulting.nl) with some printscreens and the exact setup of the widget and what is failing...

Can you do this if my proposed option fails?

Regards,
Ivo

**Comment by worthingtonte** [comment]
Hi Ivo,

Yes I have tried to use [id='[%CurrentObject%]'] and not pan to context, with no luck (I don’t see the marker). If I have pan to content set to no, and no [id='[%CurrentObject%]'] then I see the marker and the clustering of all the search locations I have used.

Ill email you some screenshots with my scenario, side by side with the GoogleMaps widget which is working for me (apart from it doesn’t have the drag location).

Thanks,

Tom
